sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -205,12 +202,18 @@
  * <em>auto commit</em> (for external monitoring or other purposes), you can set
  * <tt>"group.id"</tt>, <tt>"enable.auto.commit"</tt>, etc.
  *
- * <h3>Event Timestamp and Watermark</h3>
+ * <h3>Event Timestamps and Watermark</h3>
  * By default, record timestamp (event time) is set to processing time in KafkaIO reader and
  * source watermark is current wall time. If a topic has Kafka server-side ingestion timestamp
  * enabled ('LogAppendTime'), it can enabled with {@link Read#withLogAppendTime()}.
  * A custom timestamp policy can be provided by implementing {@link TimestampPolicyFactory}. See
  * {@link Read#withTimestampPolicyFactory(TimestampPolicyFactory)} for more information.
+ *
+ * <h3>Supported Kakfa Client Versions</h3>
+ * KafkaIO relies <i>kafka-clients</i> for all its interactions with the Kafka cluster.
+ * <i>kafka-clients</i> versions 0.10.1 and newer are supported at runtime. Please ensure that
```

Comment:
From the previous PR I understood that we were deprecating version 0.9.0.x not removing the support yet as the doc implies, is this intended? (or I am misreading).

Reply:
That is correct. I was wondering if we should mention it explicitly here that it is supported but deprecated. LMK.

Reply:
Yes I think it is worth to mention it because we still respect that contract even if deprecated.

Reply:
Agreed. Updated the JavaDoc. PTAL.